### PR TITLE
Added support for RTSP discovery in autodiscover and CLI

### PR DESCRIPTION
**Pull request by honeytung**
This PR added support for RTSP discovery in the built-in `FrameGrabber.autodiscover()` function and also can be called in CLI.

### CODE REVIEW

File: src/framegrab/cli/autodiscover.py [hunk 1]
```diff
@@ -5,21 +5,33 @@
...
+    show_default=True,
+)
+@click.option(
+    "--rtsp_discover_modes",
+    type=click.Choice(PREVIEW_RTSP_COMMAND_CHOICES, case_sensitive=False),
+    default="light",
+    show_default=True,
 )
-def autodiscover(preview: str):
+def autodiscover(preview: str, rtsp_discover_modes: str):
```
Comment (brandon-groundlight) [inline src/framegrab/cli/autodiscover.py:29]: This technically would make this a breaking change. Can we add a default to keep it compatible?

File: src/framegrab/grabber.py [hunk 2]
```diff
@@ -271,24 +272,53 @@ def create_grabber(config: dict, autogenerate_name: bool = True, warmup_delay: f
...
 
         # Autodiscover the grabbers
         grabber_list = []
         for input_type in autodiscoverable_input_types:
             logger.info(f"Autodiscovering {input_type} cameras...")
+
+            if input_type == InputTypes.RTSP:
+                onvif_devices = RTSPDiscovery.discover_onvif_devices(auto_discover_modes=rtsp_discover_modes)
+                for device in onvif_devices:
+                    for index, rtsp_url in enumerate(device.rtsp_urls):
```
Comment (brandon-groundlight) [inline src/framegrab/grabber.py:309]: This tripped me up at first glance. The idea that a single device may provide multiple valid rtsp urls that correspond to different views? That's counterintuitive, but if we have to deal with strange hardware, this is the way to do it.

**Review by brandon-groundlight — APPROVED** [review]
I have a question about devices: do we expect some devices to have multiple rtsp urls and that we may want to access different rtsp urls on the same device? I'm not sure what kinds of cameras are like that.

Other than that, one small change and then this looks good to me

File: src/framegrab/grabber.py (hunk 2 above)
Comment (honeytung) [reply src/framegrab/grabber.py:309]: RTSP camera may come with main plus sub channels so this takes into account for that. They are all the same views but main channel is the highest quality and the sub channel can sometimes be used for thumbnails.

**Pull request merged**